ouroboros-consensus/src/Ouroboros/Consensus/BlockFetchClient.hs

```diff
@@ -8,58 +8,17 @@ module Ouroboros.Consensus.BlockFetchClient
   , blockFetchClient
   ) where
 
-import           Control.Monad.Class.MonadSTM
-import           Control.Monad.Class.MonadThrow
-import           Control.Monad.Class.MonadTime
-import           Control.Tracer
-
 import           Network.TypedProtocol.Pipelined (PeerPipelined)
 import           Ouroboros.Network.Codec
 
-import           Ouroboros.Network.Block
-import           Ouroboros.Network.BlockFetch
-                     (BlockFetchConsensusInterface (..))
-import           Ouroboros.Network.BlockFetch.Client (FetchClientPolicy (..))
-import qualified Ouroboros.Network.BlockFetch.Client as BlockFetchClient
-import           Ouroboros.Network.BlockFetch.ClientState (FetchClientStateVars)
+import           Ouroboros.Network.BlockFetch.Client
+                   (blockFetchClient, FetchClientContext)
 import           Ouroboros.Network.Protocol.BlockFetch.Type
-                     (BlockFetch (BFIdle))
-
-import           Ouroboros.Consensus.Util.Condense
+                   (BlockFetch (BFIdle))
 
 -- | The block fetch layer doesn't provide a readable type for the client yet,
 -- so define it ourselves for now.
 type BlockFetchClient hdr blk m a =
```

Comment:
If we move this type synonym to `Ouroboros.Network.BlockFetch.Client`, we could even get rid of this module (`Ouroboros.Consensus.BlockFetchClient`).

Reply:
I'll let you do that as a follow-up :-)